project_task_pull_request_state/tests/test_project_task_pull_request_state.py

```diff
@@ -0,0 +1,177 @@
+
+
+class TestPullRequestState(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.Project = cls.env["project.project"]
+        cls.Task = cls.env["project.task"]
+        cls.Stage = cls.env["project.task.type"]
+
+        cls.stage_draft = cls.Stage.create({"name": "Draft"})
+        cls.stage_progress = cls.Stage.create({"name": "Progress"})
+
+        # Used to set res config settings
+        cls.res_config = cls.env["res.config.settings"]
+
+        cls.project_1 = cls.Project.create(
+            {
+                "name": "Test Project 1",
+                "type_ids": [(4, cls.stage_draft.id), (4, cls.stage_progress.id)],
+                "pr_state_default": "draft",
+            }
+        )
+        cls.project_2 = cls.Project.create(
+            {
+                "name": "Test Project 2",
+                "type_ids": [(4, cls.stage_draft.id), (4, cls.stage_progress.id)],
+                "pr_state_default": "open",
+            }
+        )
+        cls.project_3 = cls.Project.create(
+            {
+                "name": "Test Project 3",
+                "type_ids": [(4, cls.stage_draft.id), (4, cls.stage_progress.id)],
+            }
+        )
+        cls.task_1 = cls.Task.create(
+            {
+                "name": "Test Task for project 1",
+                "project_id": cls.project_1.id,
+            }
+        )
+        cls.task_2 = cls.Task.create(
+            {
+                "name": "Test Task for project 2",
+                "project_id": cls.project_2.id,
+            }
+        )
+        cls.task_3 = cls.Task.create(
+            {
+                "name": "Test Task without project",
+            }
+        )
+
+    def _set_default_pr_state(self, state):
+        """Set the default pull request state.
+
+        :param state(str): The default state for pull requests.
+
+        :return: The result of executing the record creation operation.
+        """
+        return self.res_config.create({"pr_state_default": state}).execute()
+
+    def test_pull_request_state_set_default(self):
+        """Set default PR state from project when PR URI is added to task"""
+
+        # Set default PR state
+        self._set_default_pr_state("closed")
+
+        # Set to several existing tasks at once
+        tasks = self.Task.browse([self.task_1.id, self.task_2.id, self.task_3.id])
+        tasks.write({"pr_uri": "https://@my.pr.uri/pr"})
+
+        self.assertEqual(self.task_1.pr_state, "draft", "PR state must be 'draft")
+        self.assertEqual(self.task_2.pr_state, "open", "PR state must be 'open")
+        self.assertEqual(self.task_3.pr_state, "closed", "PR state must be 'closed")
+
+        # Change pr_state
+        tasks.write({"pr_state": "closed"})
+
+        self.assertEqual(self.task_1.pr_state, "closed", "PR state must be 'closed")
```

Comment:
I guess there's closing `'` missing? If that message is even worth having

Reply:
applied